static/app/components/dropdownLink.spec.tsx

```diff
@@ -0,0 +1,314 @@
+
+        // Click again
+        userEvent.click(screen.getByText('test'));
+
+        expect(screen.getByText('test')).toBeInTheDocument();
+      });
+    });
+  });
+
+  describe('Controlled', function () {
+    describe('Opened', function () {
+      it('does not close when menu is clicked', function () {
+        render(
+          <DropdownLink title="test" alwaysRenderMenu={false} isOpen>
+            <li>hi</li>
+          </DropdownLink>
+        );
+
+        // Click option
+        userEvent.click(screen.getByText('hi'));
+
+        // Should still be open
+        expect(screen.getByText('hi')).toBeInTheDocument();
+      });
+
+      it('does not close when document is clicked', function () {
+        jest.useFakeTimers();
+
+        render(
+          <div data-test-id="outside-element">
+            <DropdownLink title="test" alwaysRenderMenu={false} isOpen>
+              <li>hi</li>
+            </DropdownLink>
+          </div>
+        );
+
+        // Click outside
+        userEvent.click(screen.getByTestId('outside-element'));
+
+        jest.runAllTimers();
+
+        // Should still be open
+        expect(screen.getByText('hi')).toBeInTheDocument();
+      });
+
+      it('does not close when dropdown actor is clicked', function () {
+        render(
+          <DropdownLink title="test" alwaysRenderMenu={false} isOpen>
+            <li>hi</li>
+          </DropdownLink>
+        );
+
+        // Click menu
+        userEvent.click(screen.getByText('test'));
+
+        // Should still be open
+        expect(screen.getByText('hi')).toBeInTheDocument();
+      });
+    });
+    describe('Closed', function () {
+      it('does not open when dropdown actor is clicked', function () {
+        render(
+          <DropdownLink title="test" alwaysRenderMenu={false} isOpen={false}>
+            <li>hi</li>
+          </DropdownLink>
+        );
+
+        // Click menu
+        userEvent.click(screen.getByText('test'));
+
+        expect(screen.queryByText('hi')).not.toBeInTheDocument();
+      });
+    });
+  });
+
+  describe('Nested Dropdown', function () {
+    const NestedDropdown = () => {
+      return (
+        <DropdownLink title="parent" alwaysRenderMenu={false}>
+          <li>
+            <DropdownLink alwaysRenderMenu={false} title="nested" isNestedDropdown>
+              <li>
+                <DropdownLink alwaysRenderMenu={false} title="nested #2" isNestedDropdown>
+                  <li>Hello</li>
+                </DropdownLink>
+              </li>
+            </DropdownLink>
+          </li>
+          <li id="no-nest">Item 2</li>
+        </DropdownLink>
+      );
+    };
+
+    it('closes when top-level actor is clicked', function () {
+      render(<NestedDropdown />);
+
+      // Open menu
+      userEvent.click(screen.getByText('parent'));
+
+      // Close menu
+      userEvent.click(screen.getByText('parent'));
+
+      expect(screen.queryByTestId('dropdown-link-menu')).not.toBeInTheDocument();
```

Comment:
would be fine if we check if there is no role `list` rendered?

Reply:
Good catch, didn't realize this had the correct roles